src/runtime/objmodel.cpp

```diff
@@ -3410,12 +3411,26 @@ Box* getiter(Box* o) {
         return r;
 
     if (typeLookup(o->cls, getitem_str, NULL)) {
-        return new BoxedSeqIter(o);
+        return new BoxedSeqIter(o, 0);
     }
 
     raiseExcHelper(TypeError, "'%s' object is not iterable", getTypeName(o)->c_str());
 }
 
+Box* getreversed(Box* o) {
+    // TODO add rewriting to this?  probably want to try to avoid this path though
+    Box* r = callattrInternal0(o, &reversed_str, LookupScope::CLASS_ONLY, NULL, ArgPassSpec(0));
+    if (r)
+        return r;
+
+    if (!typeLookup(o->cls, getitem_str, NULL)) {
+        raiseExcHelper(TypeError, "'%s' object is not iterable", getTypeName(o)->c_str());
+    }
+    int64_t len = unboxedLen(o); // this will throw an exception if __len__ isn't there
+
+    return new (seqreviter_cls) BoxedSeqIter(o, len - 1);
+}
+
```

Comment:
Tiny preference, but I would put this in the builtins.cpp file if nothing else will be calling this other than the reversed() function.